code/modules/antagonists/chrono_dude/future_tech.dm

```diff
@@ -0,0 +1,238 @@
+		return FALSE
+	return TRUE
+
+/obj/item/chrono_tele/afterattack(atom/A, mob/user, proximity)
+	if (protection_check(user))
+		. = ..()
+		do_teleport(user, A, 0, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_WORMHOLE)
+	else
+		to_chat(user, "<span class='warning'>As you try to use it, the [src] shines blue for a second, then teleports away!</span>")
+		qdel(src)
+
+/obj/item/chrono_tele/attack_self(mob/user)
+	if (protection_check(user))
+		. = ..()
+		var/atom/A = get_ranged_target_turf(get_turf(user), user.dir, 3)
+		do_teleport(user, A, 0, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_WORMHOLE)
+	else
+		to_chat(user, "<span class='warning'>As you try to use it, the [src] shines blue for a second, then teleports away!</span>")
+		qdel(src)
+
+/*
+ * CLICK REMOTE
+ */
+#define UR_PAUSE 0
+#define UR_SLOW 1
+#define UR_MUTE 2
+#define UR_STOP 3
+#define UR_MODES 4
+
+/obj/item/click_remote
+	name = "Universal Remote Control"
+	desc = "A remote able to manipulate the universe itself. Not to be used by architects."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "hand_tele"
+	item_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 3
+	throw_range = 5
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 30, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	var/mode = 0
+	var/protected = FALSE
+
+/obj/item/click_remote/proc/protection_check(mob/user)
+	if (protected && !user.mind?.has_antag_datum(/datum/antagonist/tca))
+		return FALSE
+	return TRUE
+
+/obj/item/click_remote/afterattack(atom/A, mob/user, proximity)
+	if (protection_check(user))
+		. = ..()
+		var/mob/living/carbon/target = A
+		if (target)
+			playsound(user, 'sound/machines/click.ogg', 50, TRUE, -1)
+			log_combat(user, target, "used universal remote on")
+			switch (mode)
+				if (UR_PAUSE)	//pause
+					target.Stun(500)
+					to_chat(target, "<span class='warning'>You feel like you cannot move!</span>")
+				if (UR_SLOW)	//legcuff
+					if (ishuman(target))
+						var/obj/item/restraints/legcuffs/beartrap/B = new /obj/item/restraints/legcuffs/beartrap/energy(get_turf(target))
+						B.Crossed(target)
+				if (UR_MUTE)	//mute
+					target.silent = max(500,target.silent)
+					to_chat(target, "<span class='warning'>Your voice disappeared!</span>")
+				else	//off
+					target.Sleeping(500)
+					to_chat(target, "<span class='warning'>Everything turns black...</span>")
+	else
+		to_chat(user, "<span class='warning'>As you try to use it, the [src] shines blue for a second, then teleports away!</span>")
+		qdel(src)
+
+
+/obj/item/click_remote/attack_self(mob/user)
+	if (protection_check(user))
+		. = ..()
+		mode++
+		if (mode>=UR_MODES)
+			mode = 0
```

Comment:
mode = (mode + 1) % UR_MODES

Should work pretty well here

Reply:
Thanks. Didn't know bcode supports int % total.